docker-compose/oidc/README.adoc

```diff
@@ -0,0 +1,445 @@
+
+. Clone this repository:
++
+```bash
+git clone https://github.com/redpanda-data/redpanda-labs.git
+```
+
+. Change into the `docker-compose/oidc/` directory:
++
+[,bash]
+----
+cd redpanda-labs/docker-compose/oidc
+----
+
+. Set the `REDPANDA_VERSION` environment variable to a version of Redpanda. For all available versions, see the https://github.com/redpanda-data/redpanda/releases[GitHub releases].
++
+For example:
++
+[,bash,subs="attributes+"]
+----
+export REDPANDA_VERSION=v{latest-redpanda-version}
+----
+
+. Set the `REDPANDA_CONSOLE_VERSION` environment variable to the version of Redpanda Console that you want to run. For all available versions, see the https://github.com/redpanda-data/redpanda/releases[GitHub releases].
++
+NOTE: You must use at least version v3.1.0 of Redpanda Console to deploy this lab.
++
+For example:
++
+[,bash,subs="attributes+"]
+----
+export REDPANDA_CONSOLE_VERSION=v{latest-console-version}
+----
+
+. Create a `.env` file and add the following environment variables:
++
+.`.env`
+[,bash,role="no-wrap"]
+----
+TENANT_ID=<tenant-id>
+OIDC_CLIENT_ID=<client-id>
+OIDC_CLIENT_SECRET=<client-secret>
+JWT_SIGNING_KEY=<jwt-signing-key>
+----
++
+Redpanda Console expects configuration to be passed as environment variables, where each YAML key is converted to uppercase and underscores are added for each level of nesting. For example, `authentication.oidc.clientSecret` becomes `AUTHENTICATION_OIDC_CLIENTSECRET`.
++
+To simplify this, user-friendly environment variables, such as `OIDC_CLIENT_SECRET`, are set here and then mapped to the required format in the `docker-compose.yml` file.
++
+This pattern helps separate sensitive or dynamic values from implementation-specific keys. See xref:console:config/configure-console.adoc#environment-variables[Configure Redpanda Console] for details.
+
+. Start Redpanda and Redpanda Console in Docker by running the following command:
++
+[,bash]
+----
+docker compose up --detach --wait
+----
+
+. Set up your `rpk` profile on your local machine:
++
+```bash
+rpk profile create oidc-lab --from-profile profile.yaml
+```
+
+. Update the Redpanda cluster configuration.
++
+Redpanda does not support environment variables for cluster-level properties. To configure OIDC authentication, use the following `rpk` CLI commands:
++
+[source,bash]
+----
+# Define at least one superuser. This is required for managing ACLs and cluster config.
+# Add your OIDC user’s `sub` claim here to grant superuser privileges.
+rpk cluster config set superusers '["superuser", "<oidc-sub>"]'
+
+# Enable SASL-based authentication on the Kafka API.
+rpk cluster config set enable_sasl true
+
+# Require authentication on the Admin API. Without this, the Admin API remains open.
+rpk cluster config set admin_api_require_auth true
+
+# Enable supported SASL mechanisms for the Kafka API.
+# SCRAM is for password-based users. OAUTHBEARER is for OIDC tokens.
+rpk cluster config set sasl_mechanisms '["SCRAM", "OAUTHBEARER"]'
```

Comment:
Not that it would provide much security, but wondering if the `PLAIN` mechanism is supported in this config?